### PR TITLE
feat: Replace MySQL client implementation with MySQL provider

### PR DESCRIPTION
**Pull request by sixther-dc**
#### What this PR does / why we need it:
Replace MySQL client implementation with MySQL provider.
service directory rename to apps.

#### Specified Reviewers:

/assign @luobily @skoo87 @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     Replace MySQL client implementation with MySQL provider         |
| 🇨🇳 中文    |       MySQL client的实现改为 provider      |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/dop/endpoints/application_publish_item.go [hunk 1]
```diff
@@ -20,8 +20,8 @@ import (
 	"net/http"
 
 	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/modules/apps/admin/apierrors"
```
Comment (sfwn) [inline modules/dop/endpoints/application_publish_item.go:23]: I believe this is a wrong import.

File: modules/dop/component-protocol/components/project-runtime/list/list.go [hunk 2]
```diff
@@ -36,8 +36,8 @@ import (
 	"github.com/erda-project/erda-infra/providers/component-protocol/utils/cputil"
 	"github.com/erda-project/erda/apistructs"
 	"github.com/erda-project/erda/bundle"
+	"github.com/erda-project/erda/modules/apps/personal-workbench/component-protocol/types"
```
Comment (sfwn) [inline modules/dop/component-protocol/components/project-runtime/list/list.go:39]: It's not good to use `personal-workbench`'s common types in dop side.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dop/component-protocol/components/project-runtime/list/list.go (hunk 2 above)
Comment (sfwn) [reply modules/dop/component-protocol/components/project-runtime/list/list.go:39]: There are many other places with the same problem.

**Review by sfwn — CHANGES_REQUESTED** [review]
dingtalktest should not be an app. It's a functional provider.
What do you think @luobily @Effet

**Comment by sixther-dc** [comment]
> 
these code will be deleted, next pr.

**Review by sfwn — APPROVED** [review]

**Pull request merged**